resources/todomvc/architecture-examples/preact/src/app/app.js

```diff
@@ -0,0 +1,69 @@
+// eslint-disable-next-line no-unused-vars
+import { h } from "preact";
+import { useEffect, useState } from "preact/hooks";
+
+import TodoModel from "./model";
+import TodoHeader from "./header";
+import TodoMain from "./main.js";
+import TodoFooter from "./footer";
+import { FILTERS } from "./utils";
+
+const getRoute = () => {
+    let route = String(location.hash || "")
+        .split("/")
+        .pop();
+
+    // prettier-ignore
+    if (!FILTERS[route])
+        route = "all";
+
+    return route;
+};
+
+export default function App() {
+    const [, setUdpatedAt] = useState(Date.now());
+    const [route, setRoute] = useState("all");
+
+    function update() {
+        setUdpatedAt(Date.now());
+    }
+
+    const model = TodoModel(update);
+
+    useEffect(() => {
+        function handleHashChange() {
+            setRoute(getRoute());
+            update();
```

Comment:
My understanding is that the `update` is unnecesary here because `setRoute` will already retrigger a render.

Reply:
removed - thanks 😄